modelskill/comparison/_utils.py

```diff
@@ -84,6 +86,52 @@ def calc_metrics(x):
     return res
 
 
+def _dt_in_by(by):
+    by = [by] if isinstance(by, str) else by
+    if any(str(by).startswith("dt:") for by in by):
+        return True
+    return False
+
+
+ALLOWED_DT = [
+    "year",
+    "quarter",
+    "month",
+    "month_name",
+    "day",
+    "day_of_year",
+    "dayofyear",
+    "day_of_week",
+    "dayofweek",
+    "hour",
+    "minute",
+    "second",
+    "weekday",
+]
+
+
+def _add_dt_to_df(df, by):
+    ser = df.index.to_series()
+    by = [by] if isinstance(by, str) else by
```

Comment:
Is it necessary to handle `by` as a str | List[str] in several places, would be nice to do it once.

Reply:
Makes sense, I guess I was not sure about whether it makes a difference in the pandas method df.groupby(by=) to have e.g. "model" vs ["model"]. But I should test (just a bit stressed right now)